integration/test/Test/LegalHold.hs

```diff
@@ -274,3 +280,522 @@ testLHDeleteClientManually = do
+
+      legalholdWhitelistTeam tid alice
+        >>= assertStatus 200
+
+      -- the status messages for these have already been tested
+      postLegalHoldSettings tid alice (mkLegalHoldSettings lhPort)
+        >>= assertStatus 201
+
+      -- legalhold has been requested but is disabled
+      statusShouldbe "disabled"
+
+      requestLegalHoldDevice tid alice bob
+        >>= assertStatus 201
+
+      -- legalhold has been set to pending after requesting device
+      statusShouldbe "pending"
+
+      approveLegalHoldDevice tid bob defPassword
+        >>= assertStatus 200
+
+      -- bob has accepted the legalhold device
+      statusShouldbe "enabled"
+
+type TB s = TaggedBool s
+
+ut :: TB s -> Bool
+ut = unTagged
+
+testLHNoConsentBlockOne2OneConv :: TB "connect first" -> TB "team peer" -> TB "approve LH" -> TB "test pending connection" -> App ()
+testLHNoConsentBlockOne2OneConv
+  (ut -> connectFirst)
+  (ut -> teampeer)
+  (ut -> approveLH)
+  (ut -> testPendingConnection) =
+    startDynamicBackends [mempty] \[dom1] -> do
+      -- team users
+      -- alice (team owner) and bob (member)
+      (alice, tid, []) <- createTeam dom1 1
+      bob <-
+        if teampeer
+          then do
+            (walice, _tid, []) <- createTeam dom1 1
+            -- FUTUREWORK(mangoiv): creating a team on a second backend
+            -- causes this bug: https://wearezeta.atlassian.net/browse/WPB-6640
+            pure walice
+          else randomUser dom1 def
+
+      legalholdWhitelistTeam tid alice
+        >>= assertStatus 200
+
+      let doEnableLH :: HasCallStack => App (Maybe String)
+          doEnableLH = do
+            -- alice requests a legalhold device for herself
+            requestLegalHoldDevice tid alice alice
+              >>= assertStatus 201
+
+            when approveLH do
+              approveLegalHoldDevice tid alice defPassword
+                >>= assertStatus 200
+            legalholdUserStatus tid alice alice `bindResponse` \resp -> do
+              resp.status `shouldMatchInt` 200
+              resp.json %. "status" `shouldMatch` if approveLH then "enabled" else "pending"
+            if approveLH
+              then do
+                aliceId <- objId alice
+                BrigI.getClientsFull alice [aliceId] `bindResponse` \resp -> do
+                  [lhd] <-
+                    resp.json %. aliceId
+                      & asList
+                      >>= filterM \val -> (== "legalhold") <$> (val %. "type" & asString)
+                  lhdid <- lhd %. "id" & asString
+                  pure (Just lhdid)
+              else pure Nothing
+
+          doDisableLH :: HasCallStack => App ()
+          doDisableLH =
+            disableLegalHold tid alice alice defPassword
+              >>= assertStatus 200
+
+      withMockServer lhMockApp \lhPort _chan -> do
+        postLegalHoldSettings tid alice (mkLegalHoldSettings lhPort)
+          >>= assertStatus 201
+
+        if not connectFirst
```

Comment:
why `not`?

Reply:
I also stumbled over this in the old test but I *does* make sense.